### PR TITLE
feat: rebase_on_cloudposse

### PR DESCRIPTION
**Pull request by mfroembgen**

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -2,37 +2,45 @@ provider "aws" {
...
   nat_instance_enabled = false
 
   context = module.this.context
 }
 
 resource "aws_ecs_cluster" "default" {
-  name = module.this.id
-  tags = module.this.tags
+  #bridgecrew:skip=BC_AWS_LOGGING_11: not required for testing
+  count = local.enabled ? 1 : 0
```
Comment (applike-ss) [inline examples/complete/main.tf:34]: what's the point for this in an example? 🤔

File: examples/complete/main.tf [hunk 2]
```diff
@@ -45,11 +53,34 @@ module "container_definition" {
...
+
+  context = module.this.context
+}
+
 module "ecs_alb_service_task" {
   source                             = "../.."
   alb_security_group                 = module.vpc.vpc_default_security_group_id
-  container_definition_json          = module.container_definition.json_map_encoded_list
-  ecs_cluster_arn                    = aws_ecs_cluster.default.arn
+  container_definition_json          = one(module.container_definition[*].json_map_encoded_list)
```
Comment (applike-ss) [inline examples/complete/main.tf:82]: afaik using `*` as an index selector is deprecated or bad pratice or something. We've settled on using numeric indices with try everywhere by now. Please don't break this.

File: main.tf [hunk 3]
```diff
@@ -457,9 +465,15 @@ resource "aws_ecs_service" "ignore_changes_task_definition" {
     }
   }
 
+  triggers = local.redeployment_trigger
```
Comment (applike-ss) [inline main.tf:468]: How about renaming this to triggers instead, should there ever be other triggers?
```suggestion
  triggers = local.triggers
```

File: main.tf [hunk 4]
```diff
@@ -457,9 +465,15 @@ resource "aws_ecs_service" "ignore_changes_task_definition" {
...
+
   lifecycle {
     ignore_changes = [task_definition]
   }
+
+  # Avoid race condition on destroy.
+  # See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service
+  depends_on = [aws_iam_role.ecs_service, aws_iam_role_policy.ecs_service]
```
Comment (applike-ss) [inline main.tf:476]: should we add this to our redis/kvstore module as well?

**Review by applike-ss — CHANGES_REQUESTED** [review]

File: examples/complete/main.tf (hunk 1 above)
Comment (mfroembgen) [reply examples/complete/main.tf:34]: You need to ask cloudposse about that ^^

File: examples/complete/main.tf (hunk 2 above)
Comment (mfroembgen) [reply examples/complete/main.tf:82]: That is exactly what one is for and why couldposse uses it here.

File: main.tf (hunk 3 above)
Comment (mfroembgen) [reply main.tf:468]: Again I would leave it like that because we keep the repo as close as possible from the fork

File: main.tf (hunk 4 above)
Comment (mfroembgen) [reply main.tf:476]: Yes, but I have something different in mind for this modules.  I will create a ticket for it.

**Review by applike-ss — APPROVED** [review]

**Pull request merged**